astropy/cosmology/parameters.py

```diff
@@ -150,4 +144,5 @@
                "Table 1 (WMAP + BAO + SN ML).")
 )
 
-available = tuple(k for k in locals() if not k.startswith('_'))
+# If new parameters are added, this list must be updated
```

Comment:
Looking at this again, why not keep the autogeneration, but just ensure it works? E.g.,
```
available = [k for k in locals() if isinstance(k, dict) and 'reference' in k]
```
(I'd also remove the explicit addition from `core.__all__` and instead write there `__all__ = [...] + parameters.available` or add them in the loop further down, but that is a bit beyond this PR...)